### PR TITLE
move build configuration to `pyproject.toml`

### PR DESCRIPTION
**Pull request by zacharyburnett**
`setuptools` now supports the `[project]` table, which is defined by [PEP621](https://peps.python.org/pep-0621).

Additionally, `setuptools` now supports its own entry in `pyproject.toml` called `[tool.setuptools]` (https://github.com/pypa/setuptools/issues/1688, https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#setuptools-specific-configuration); however, it comes with the following disclaimer:
> Support for declaring configurations not standardized by [PEP 621](https://peps.python.org/pep-0621/) (i.e. the `[tool.setuptools]` table), is still in **beta** stage and might change in future releases.

Reading `toml` is supported natively in Python 3.11 with `tomllib`

Given this, we can attempt to consolidate the build configuration into a single `pyproject.toml` file that can possibly be read by other build systems in the future.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -14,23 +14,19 @@
 
 import datetime
 import importlib
-import sys
 import os
 import sphinx
 import stsci_rtd_theme
+import sys
+import tomli
```
Comment (zacharyburnett) [inline docs/conf.py:21]:
```suggestion
if sys.version_info < (3, 11):
     import tomli as tomllib
 else:
     import tomllib
```

File: docs/conf.py [hunk 2]
```diff
@@ -40,8 +36,9 @@ def setup(app):
 sys.path.insert(0, os.path.abspath('exts/'))
 
 # -- General configuration ------------------------------------------------
-conf.read([os.path.join(os.path.dirname(__file__), '..', 'setup.cfg')])
-setup_cfg = dict(conf.items('metadata'))
+with open(Path(__file__).parent.parent / "pyproject.toml", "rb") as configuration_file:
+    conf = tomli.load(configuration_file)
```
Comment (zacharyburnett) [inline docs/conf.py:40]:
```suggestion
    conf = tomllib.load(configuration_file)
```

**Comment by mfixstsci** [comment]
@zacharyburnett I noticed and am updating some of the docs/removing deprecated test runs etc. I am hoping to have it in better shape this afternoon!

**Comment by pllim** [comment]
Are you sure this PR is complete?

`E   ModuleNotFoundError: No module named 'requests'`

Should not merge PR with red CI, if possible.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    "scipy",
+    "tomli; python_version<\"3.11\"",
+]
+
+[build-system]
+requires = [
+    "setuptools>=61.2",
+    "setuptools_scm[toml]>=3.4",
+    "wheel",
+    "oldest-supported-numpy",
```
Comment (pllim) [inline pyproject.toml:57]: Are you sure you need oldest-supported-numpy . Do you have C-extension that uses numpy ABI?

Comment (mfixstsci) [reply pyproject.toml:57]: @zacharyburnett I am assuming this a hold over from the `setup.py` file. I can double check and open a new PR with a fix.

**Comment by mfixstsci** [comment]
> Are you sure this PR is complete?
> 
> `E ModuleNotFoundError: No module named 'requests'`
> 
> Should not merge PR with red CI, if possible.

I agree, but sadly the tests have been failing for some time now. There are some issues that are instrument specific that cause them which are out of @Witchblade101 and I's expertise. In this case I was wrong and should have double checked. I can

File: pyproject.toml (hunk 3 above)
Comment (mfixstsci) [reply pyproject.toml:57]: Looks like we versioned `numpy` in the `setup.py` file. Any reason for using the "oldest-supported-version" of numpy?

Comment (pllim) [reply pyproject.toml:57]: This section is for *build*, not *install*. Do you need numpy during compile time?

Comment (mfixstsci) [reply pyproject.toml:57]: Ah, right sorry. I don't believe we do!

**Comment by mfixstsci** [comment]
> > Are you sure this PR is complete?
> > `E ModuleNotFoundError: No module named 'requests'`
> > Should not merge PR with red CI, if possible.
> 
> I agree, but sadly the tests have been failing for some time now. There are some issues that are instrument specific that cause them which are out of @Witchblade101 and I's expertise. In this case I was wrong and should have double checked.

@pllim maybe it was an issue with python 3.7 because @zacharyburnett just added another PR #319 and the test are all passing now 🎉

**Pull request merged**